app/src/main/java/com/example/sharingang/UserProfileFragment.kt

```diff
@@ -162,22 +179,44 @@ class UserProfileFragment : Fragment() {
         when (button) {
             binding.btnOpenCamera, binding.btnOpenGallery -> {
                 binding.applyholder.visibility = View.VISIBLE
-                if(button == binding.btnOpenGallery) imageAccess.openGallery()
+                if (button == binding.btnOpenGallery) imageAccess.openGallery()
                 else imageAccess.openCamera()
             }
             binding.btnApply -> {
                 imageUri = imageAccess.getImageUri()
                 if (imageUri != Uri.EMPTY && imageUri != null) {
                     binding.imageView.setImageURI(imageUri)
                     lifecycleScope.launch(Dispatchers.IO) {
-                        userRepository.add(userRepository.get(currentUserId!!)!!.copy(profilePicture = imageUri.toString()))
+                        userRepository.add(
+                            userRepository.get(currentUserId!!)!!
+                                .copy(profilePicture = imageUri.toString())
+                        )
                     }
                 }
                 binding.applyholder.visibility = View.GONE
             }
         }
     }
 
+    private fun setupReportButton() {
+        var username = ""
+        if (currentUserId != null) {
+            binding.btnReport.visibility = View.VISIBLE
+            lifecycleScope.launch(Dispatchers.IO) {
+                username = userRepository.get(shownUserProfileId!!)!!.name
+            }
+        }
+
+        binding.btnReport.setOnClickListener {
+            userViewModel.report(shownUserProfileId!!, currentUserId!!)
+            val builder = AlertDialog.Builder(requireContext())
+            builder.setTitle("Report Result")
+            builder.setMessage("Successfully reported $username.")
```

Comment:
Last thing and I'm sorry, but can you make it so the report function returns the username that way you don't have to user another coroutine in the fragment to get it. Just get it in the viewModel while you report and then return the username. I can take care of that another day if you want though.